spec/features/user_can_add_items_to_cart_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+
+feature "cart" do
+  fixtures :categories
+  fixtures :items
+  fixtures :users
+
+  before(:each) do
+    visit "/"
+  end
+
+  it "can visit the cart page" do
+    click_link("Cart")
+    expect(current_path).to eq('/cart')
+  end
+
+  it "can add items" do
+    click_link("All Products")
+    first(:button, "Add To Cart").click
+    expect(current_path).to eq('/cart')
+    expect(page).to have_content("Item A")
+    expect(page).to have_content("$11.11")
+  end
+
+  context "user is logged in" do
+    before(:each) {
+      visit '/login'
+      fill_in "Email", with: "jeff@gmail.com"
+      fill_in "Password", with: "pass"
+      click_button("Login")
+    }
+
+    it "can checkout to buy items" do
+      click_link("All Products")
+      first(:button, "Add To Cart").click
+      click_button("Checkout")
+
+      expect(current_path).to eq("/checkout")
+      expect(page).to have_content("Item A")
+      expect(page).to have_content("$11.11")
+      expect(page).to have_content("Total")
+    end
+  end
+
+  context "user is not logged in" do
+
+    it "must log in before checking out" do
+      visit '/'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.